efinance/utils/__init__.py

```diff
@@ -133,36 +156,55 @@ def search_quote(
     """
     # NOTE 本地仅存储第一个搜索结果
     if use_local and count == 1:
-        quote = search_quote_locally(keyword)
+        quote = search_quote_locally(keyword, market_type=market_type)
         if quote:
             return quote
     url = 'https://searchapi.eastmoney.com/api/suggest/get'
     params = (
         ('input', f'{keyword}'),
         ('type', '14'),
         ('token', 'D43BF722C8E33BDC906FB84D85E326E8'),
-        ('count', f'{count}'),
+        ('count', f'{max(count, 5)}'),
     )
-    json_response = session.get(url, params=params).json()
-    items = json_response['QuotationCodeTable']['Data']
-    if items is not None:
-        quotes = [Quote(*item.values()) for item in items]
+    try:
+        json_response = session.get(url, params=params).json()
+        items = json_response['QuotationCodeTable']['Data']
+    except json.JSONDecodeError:
+        RuntimeWarning("unable to parse search quote result, consider if you are blocked")
+        return None
+
+    if items is not None and items:
+        quotes = [
+            Quote(*item.values())
+            for item in items
+            # 支持精确查找股票代码
+            if (not kwargs.get(MagicConfig.QUOTE_SYMBOL_MODE, False) or (keyword == item['Code'])) 
+                # 支持筛选股票市场
+                and (market_type is None or (market_type.value == item['Classify']))
+        ]
         # NOTE 暂时仅存储第一个搜索结果
         save_search_result(keyword, quotes[:1])
         if count == 1:
-            return quotes[0]
-        return quotes
+            return quotes[0] if len(quotes) >= 1 else None
+
+        return quotes[:count + 1]
```

Comment:
It should be `return quotes[:count]`

Reply:
See [correct condition of multiple returns of quote info](https://github.com/Micro-sheep/efinance/pull/161/commits/3b96e43c3ab16f53cc7e4d9f718f47dd0928739e)